.github/workflows/label-cherry-pick.yaml

```diff
@@ -0,0 +1,71 @@
+name: Label Cherry Pick
+
+on:
+  pull_request_target:
+    types:
+      - labeled
+      - closed
+
+jobs:
+  cherry_pick:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target_branch: ['23.lts.1+', '22.lts.1+', '21.lts.1+', '20.lts.1+', '19.lts.1+', '19.android.1+', 'rc_11', 'COBALT_9']
+    if: |
+      github.event.pull_request.merged == true &&
+      github.event.pull_request.merge_commit_sha != null
+    env:
+      ACCESS_TOKEN: ${{ secrets.CHERRY_PICK_TOKEN }}
+      REPOSITORY: ${{ github.repository }}
+      GITHUB_REF: ${{ github.ref }}
+      MERGE_COMMIT_SHA: ${{ github.event.pull_request.merge_commit_sha }}
+    steps:
+      - name: Default RUN_FOR_CURRENT_BRANCH to false
+        run: echo "RUN_FOR_CURRENT_BRANCH=false" >> $GITHUB_ENV
+
+      - name: Set RUN_FOR_CURRENT_BRANCH for closed pull_request
+        if: |
+          github.event.action == 'closed' &&
+          github.base_ref != matrix.target_branch &&
+          contains(github.event.pull_request.labels.*.name, format('cp-{0}', matrix.target_branch))
+        run: echo "RUN_FOR_CURRENT_BRANCH=true" >> $GITHUB_ENV
+
+      - name: Set RUN_FOR_CURRENT_BRANCH for labeled pull_request
+        if: |
+          github.event.action == 'labeled' &&
+          github.base_ref != matrix.target_branch &&
+          github.event.label.name == format('cp-{0}', matrix.target_branch)
+        run: echo "RUN_FOR_CURRENT_BRANCH=true" >> $GITHUB_ENV
```

Comment:
Since this must variable must be true for the rest of the steps to execute would it make sense to `exit 1` to fail if it's not? If you don't there's nothing happens which might be a little confusing for the invoker.

Reply:
Found a nicer alternative to `exit 1` using github scripts: `core.setFailed`

https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions

Reply:
But the job didn't fail, it just didn't run for that branch—I wouldn't really want to mark it as a failure in that case

Reply:
Ok, I thought that you would trigger the job for a specific branch. This makes more sense then.

I tried to find a way to exit a job early but that functionality doesn't seem to exist.

Reply:
This should be cleaner now